src/ast/mod.rs

```diff
@@ -7317,6 +7367,87 @@ impl Display for UtilityOption {
     }
 }
 
+/// MSSQL's `JSON_ARRAY` constructor function
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub struct JsonArray {
+    pub args: Vec<Expr>,
+    pub null_clause: Option<JsonNullClause>,
+}
+
+impl Display for JsonArray {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "JSON_ARRAY({}", display_comma_separated(&self.args))?;
+        if let Some(null_clause) = &self.null_clause {
+            if !self.args.is_empty() {
+                write!(f, " ")?;
+            }
+            write!(f, "{null_clause}")?;
+        }
+        write!(f, ")")
+    }
+}
+
+/// MSSQL's `JSON_OBJECT` constructor function
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
```

Comment:
Ah I think piggy-backing on my comment on the JsonArray, we would probably support this syntax automatically  by extending the function_arguments_list representation? Did you see the existing [json functionality](https://github.com/apache/datafusion-sqlparser-rs/blob/a4c73f3c6b221d4db51c8126330e177c5d9bd820/src/parser/mod.rs#L2951), I would expect the parser currently supports `foo:bar` as an expression today iirc that's snowflake json syntax, which is identical to this and feels like should be supported already?